File: essentiel-app/src/screens/WeekdaysChecker.js
import React from 'react';
import { View, Text } from 'react-native';
import FontAwesome5Icon from 'react-native-vector-icons/FontAwesome5';

const CustomWeekdays = ({ workoutDays }) => {
  // Defines the weekdays starting from Monday
  const weekdays = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'];

  return (
    <View style={{ flexDirection: 'row', justifyContent: 'space-around' }}>
      {weekdays.map((day, index) => {
        // Direct mapping of workoutDays to weekdays, no adjustment needed
        const isChecked = workoutDays[index] !== 0; // Check if the corresponding day is marked as a workout day

        return (
          <View key={day} style={{ alignItems: 'center' }}>
            <FontAwesome5Icon
              name={isChecked ? 'check-square' : 'square'} // Display check-square if workout day, else square
              color={'white'}
              size={26}
              solid={isChecked} // Use solid icon if it's a workout day
            />
            <Text style={{ fontSize: 18, color: 'white' }}>{day}</Text>
          </View>
        );
      })}
    </View>
  );
};

export default CustomWeekdays;

File: lib/components/WeekdaysChecker.dart
import 'package:flutter/material.dart';

class CustomWeekdays extends StatelessWidget {
  final List<bool> workoutDays; // List of booleans indicating workout days

  CustomWeekdays({required this.workoutDays, Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    final List<String> weekdays = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'];

    return Row(
      mainAxisAlignment: MainAxisAlignment.spaceAround,
      children: List.generate(weekdays.length, (index) {
        bool isChecked = workoutDays[index]; // Determine if the day is a workout day

        return Column(
          children: [
            Icon(
              isChecked ? Icons.check_box : Icons.check_box_outline_blank, // Material icons
              color: isChecked ? Colors.white : Colors.grey[600],
              size: 26,
            ),
            SizedBox(height: 4),
            Text(
              weekdays[index],
              style: TextStyle(fontSize: 18, color: Colors.grey[600], fontWeight: FontWeight.bold),
            ),
          ],
        );
      }),
    );
  }
}
